### ISSUE TITLE
Investigate options for obofoundry.org resolving to new site

### ISSUE THREAD

**Issue by cmungall** [OBOFoundry/OBOFoundry.github.io#29]
cc @alanruttenberg @kltm

Can we just use a CNAME?

**Comment by cmungall** [comment]
Shall we just redirect while we investigate?

**Comment by cmungall** [comment]
will talk with @kltm today

**Comment by alanruttenberg** [comment]
github documents how to set up the domain stuff at https://help.github.com/articles/setting-up-a-custom-domain-with-github-pages/

The instruction is to use a CNAME in the obofoundry domain (I am administrator until we can move it to a shared account) and to put a file in the repository that lets github know what domain you are serving.

**Comment by alanruttenberg** [comment]
Do we want obofoundry.org or obolibrary.org 
I'd argue for the latter with the former having some information that is specific to the foundry and a link to the library.

**Comment by kltm** [comment]
Okay, have gotten my hands dirty here a little bit. The MX records look fine for whatever we'd want to do, so we're left with the A record for address 54.83.197.87.
Unfortunately, the LBL DNS interface is less than clear on how to accomplish the tasks necessary for what github needs:
https://help.github.com/articles/tips-for-configuring-an-a-record-with-your-dns-provider/#configuring-an-a-record-with-your-dns-provider
I'll go ahead and put in a help request LBL just to see what would need to be done.

**Comment by kltm** [comment]
To quote an email with the LBL people earlier today:
```
My understanding is that in the second case (www.obofoundry.org as a
CNAME to obofoundry.github.io), obofoundry.org would not resolve, so
that's wouldn't work.

The problem with the A record is that it cannot be to an alias (if I
understand this all right); but we do not have a static IP address that
we can use (which is why the github instructions talk about the DNS
changes), so that won't work either.

It sounds like LBL might not have the setup to do the changes as needed
for this setup?

I think it might be best to revert obofoundry.org to:

 obofoundry.org.		8538	IN	A	54.83.197.87

and we can take care of it with a proxypass at the server level until we
have another plan.
```
I've done the proxy pass, but there are two wrinkles.

The first is that github.io defaults to https, and the proxy is not, so that's weird, but livable. However, an issue comes up if somebody tries to use the CNAME file in the github repo: while the main pages seem fine, some of the assets cannot be found with the redirect. This may have something to do with the https proxying, but I haven't really dug in (already have spent too much time on this today).

**Comment by cmungall** [comment]
Let's call this closed. Going to obofoundry.org takes you to the new site, thanks!

**Comment by alanruttenberg** [comment]
@ktlm Some background. root domains are generally not advised or not supported by registrars. Not by Google, for example. My bad. Instead there are three options.

1. The first, the one I tested, relies on some Github magic. You create a CNAME for www.obofoundry.org and neither a CNAME nor an ANAME for obofoundry.org. Then you create a file called CNAME in the root directory of the project. Then the magic arranges for the root domain to forward to the www domain.

See https://github.com/OBOFoundry/OBOFoundry.github.io/commit/ce83d38371772eba3d8da4444e8c3ac25f00b236 where I tried this.

That is what I recommend.

2. An alternative is to use subdomain forwarding. See http://serverfault.com/questions/617248/does-google-domains-support-cname-like-functionality-at-the-zone-apex#answer-714357 I don't know if LBL's DNS supports that, but google domains does.

3. Another way is described in the post below. It relies on a custom record: 'ANAME', 'ALIAS'...

http://stackoverflow.com/questions/656009/how-to-overcome-root-domain-cname-restrictions#answer-22659895

Note: I think that we should have this page live at obolibrary.org rather than obofoundry.org and have obofoundry.org have information about the foundry specifically, and a link to obolibrary.org. I will submit another ticket about that.
